feat: Add "Goals for all crates" section

winksaville opened the pull request:
Initial proposal for goals for all crab-cab crates

winksaville committed:
feat: Add "Goals for all crates" section

Initial proposal for goals for all crab-cab crates

TimTheBig commented:
Thank you for the pr

TimTheBig commented on README.md:
```diff
@@ -1,3 +1,27 @@
+## Goals for all crates
+
+- Minimum Supported Rust Version (MSRV)
+  - 1.85.1 Edition 2024
```
This should stay under polices in the description, maybe add polices to the plan

winksaville replied:
I've updated the commit removing mentioning MSRV

winksaville committed:
feat: add Policies for all crates

Moved `cargo commands` to Policy.

Obviously, what is a Goal vs Policy is up for debate

winksaville commented:
This morning I saw [this discussion](https://internals.rust-lang.org/t/array-map-annotate-with-must-use/22813/10) and we might want a list of lints and attributes in our goal or policy sections.

TimTheBig commented:
> This morning I saw [this discussion](https://internals.rust-lang.org/t/array-map-annotate-with-must-use/22813/10) and we might want a list of lints and attributes in our goal or policy sections.

Smart, I like to use:
```java
#![forbid(unsafe_code)]
#![deny(unused, clippy::missing_const_for_fn)]
#![warn(clippy::all)]
```
I think some from `clippy::pedantic` should also be included. I like to use `clippy::pedantic` as a tool when refactoring, not fixing all.
